lib/Sema/TypeCheckConstraints.cpp

```diff
@@ -451,8 +451,20 @@ resolveDeclRefExpr(UnresolvedDeclRefExpr *UDRE, DeclContext *DC) {
 
     performTypoCorrection(DC, UDRE->getRefKind(), Type(), Name, Loc,
                           lookupOptions, Lookup);
+    
+    // Check if any lookups share a basename. If that's the case, we need
+    // to disambiguate the diagnostic by printing the UDRE's full name.
```

Comment:
Ah, that part. Sorry, I didn't realize that's what you were referring to. We added that rule because of imported C functions having a zillion unnamed parameters.